cardano-testnet/test/Test/Cli/QuerySlotNumber.hs

```diff
@@ -0,0 +1,157 @@
+import qualified Data.Time.Format as DT
+import           GHC.Stack (HasCallStack, withFrozenCallStack)
+import           Hedgehog (Property)
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.Process as H
+import qualified Hedgehog.Internal.Property as H
+import           Prelude
+import qualified System.Directory as IO
+import           System.Environment (getEnvironment)
+import           System.FilePath ((</>))
+import qualified System.Info as SYS
+import           Text.Read
+
+import           Cardano.Testnet
+import           Testnet.Babbage (startTimeOffsetSeconds)
+import qualified Testnet.Util.Base as H
+import           Testnet.Util.Process
+import           Testnet.Util.Runtime
+
+hprop_querySlotNumber :: Property
+hprop_querySlotNumber = H.integrationRetryWorkspace 2 "query-slot-number" $ \tempAbsBasePath' -> do
+  H.note_ SYS.os
+  base <- H.note =<< H.noteIO . IO.canonicalizePath =<< H.getProjectBase
+  configurationTemplate <- H.noteShow $ base </> "configuration/defaults/byron-mainnet/configuration.yaml"
+  conf@Conf { tempBaseAbsPath } <- H.noteShowM $
+    mkConf (ProjectBase base) (YamlFilePath configurationTemplate) tempAbsBasePath' Nothing
+
+  let
+    testnetOptions = BabbageOnlyTestnetOptions $ babbageDefaultTestnetOptions
+      { babbageNodeLoggingFormat = NodeLoggingFormatAsJson
+      }
+  tr@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    } <- testnet testnetOptions conf
+  ShelleyGenesis{sgSystemStart, sgSlotLength} <- H.noteShowM $ shelleyGenesis tr
+
+  -- there's an offset for start time, so we have to account for that here
+  let startTime = ((-1) * startTimeOffsetSeconds) `DT.addUTCTime` sgSystemStart
+      slotLength = fromNominalDiffTimeMicro sgSlotLength
+      -- how many slots can the checked value differ from
+      -- we have 1s precision for UTC timestamp CLI argument, so this value tells how many slots in 1s can be
+      slotPrecision = round $ 1 / slotLength
+      -- first era has epochsize 100 - that's a different value than in ShelleyGenesis
+      epochSize = 100
+      passedSlots = 100 :: Int
+      passedTime = fromIntegral passedSlots * fromNominalDiffTimeMicro sgSlotLength
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  env <- H.evalIO getEnvironment
+  execConfig <- H.noteShow H.ExecConfig
+    { H.execConfigEnv = Last $ Just $
+      [ ("CARDANO_NODE_SOCKET_PATH", IO.sprocketArgumentName poolSprocket1)
+      ]
+      -- The environment must be passed onto child process on Windows in order to
+      -- successfully start that process.
+      <> env
+    , H.execConfigCwd = Last $ Just tempBaseAbsPath
+    }
+
+  -- {{{ Test cases
+  id do
+    H.note_ "Double check that start time offset is larger than the passed time we test"
+    H.diff startTimeOffsetSeconds (>) passedTime
```

Comment:
Further tests rely on this assumption. So this is just an early warning that `startTimeOffsetSeconds`  was modified and has a different value than this test is expecting. Otherwise we would just get a failure down the line, which would require some investigation.